src/dmd/dtemplate.d

```diff
@@ -1928,6 +1928,10 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
                             {
                                 // Allow conversion from T[lwr .. upr] to ref T[upr-lwr]
                             }
+                            else if (global.params.rvalueRefParam)
+                            {
+                                // Allow implicit conversion to ref
```

Comment:
Wait hold on, What type of implicit conversion are we talking about here? Non-const Ref parameters shouldn't accept rvalues from implicit conversions here!

Reply:
```
ref int foo();
void goo(int);
void test()
{
    goo(foo());
}
```
with `-preview=rvaluerefparam` they do and should.

Reply:
I am talking about a case scenario involving this:
```int x = 0;
void goo(ref long x) { x = 1} ;
void test()
{
    goo(x); /// a rvalue that is the result of a implicit conversion from an lvalue, which is a bad idea.
    assert(x == 1); // This assertion will fail.
}```
This has brought up regarding the rejection of the rvalue dip here.
https://forum.dlang.org/post/q2pcjn$15nq$1@digitalmars.com

Reply:
Note that this only affects the template deduction portion of the overall validity checking, there is still the function argument matching which will still reject this.

Reply:
Maybe this test should be added to make sure something doesn't brake.

```d
void goo()(ref long x) { x = 1; }
void test()
{
    int x;
    // this should fail.
    goo(x); // a rvalue that is the result of a implicit conversion from an lvalue, which is a bad idea.
    assert(x == 1);
}
```